src/util/rotary.h

```diff
@@ -1,40 +1,15 @@
 #pragma once
 
-#include <vector>
+#include <QQueue>
 
+// Simple moving average
 class Rotary {
   public:
-    Rotary();
-
-    // Start calibration measurement
-    void calibrateStart();
-    // End calibration measurement
-    double calibrateEnd();
-    // Set calibration
-    void setCalibration(double c);
-    // Get calibration
-    double getCalibration();
+    Rotary(qsizetype filterLength)
+            : m_filterHistory(QList<double>(filterLength, 0.0)){};
     // Low pass filtered rotary event
-    double filter(double dValue);
-    // Hard set event value
-    double fillBuffer(double dValue);
-    // Collect calibration data
-    void calibrate(double dValue);
-    // Set filter length
-    void setFilterLength(int i);
-    // Get filter length
-    int getFilterLength();
+    double filter(double value);
 
-  protected:
-    // Length of filter
-    int m_iFilterLength;
-    // Update position in filter
-    int m_iFilterPos;
-    // Pointer to rotary filter buffer
-    std::vector<double> m_pFilter;
-    // Calibration value
-    double m_dCalibration;
-    // Last value
-    double m_dLastValue;
-    int m_iCalibrationCount;
+  private:
+    QQueue<double> m_filterHistory;
```

Comment:
QQueue is not effective for this prupose. I suggest to restore the `std::vector<double>` based solution. 
The `m_pFilter` can be called `m_pRingBuffer` to make the use cas obvious.

Reply:
yes, I did something similar in a follow up with I haven't pushed yet.

Reply:
see 7578cfaf14fcb054f4f8dd210fcabcd2a30f1c7c